NegaMax.java

```diff
@@ -1,72 +1,88 @@
+	//Don't think best-move is necessary now that I've changed the way the negamax works, but I'm gonna leave it for now
+	//private static MoveAndValue best_move;
+	private static int depth;
+	private static int alpha;
+	private static int beta;
+	private static int color;
 
 	//runs the negaMax and returns the best move
-	public int nextMove(Game game){
+	public static Move nextMove(Board game){
 		depth = 0;
 		alpha = 1000;
 		beta = -1000;
 		color = 1;
 		int startTime = 0; /*This will be the start time of the negamax*/
 		int depthCount = 1;
 		//iterative deepening
-		MoveAndValue [] moves = game.getAvailableMoves();
+		List<Move> tempMoves = game.generateMoves();
+		MoveAndValue [] moves = new MoveAndValue[tempMoves.size()];
+		int count = 0;
+		for(Move m : tempMoves){
+			MoveAndValue tempMove = new MoveAndValue(m,-1);
+			moves[count] = tempMove;
+		}
+
 		while(10 /*this should be current time*/ - startTime < 10){
 			for(int i=0; i<moves.length; i++){
-				best_move.move = -1;
+				//Don't think this is necessary, but I'm gonna leave it for now
+				/*
+				best_move.move = null;
 				best_move.value = -1;
+				*/
 				game.makeMove(moves[i].move);
 				//Note: may have to make alpha and beta instance variables so that the iterative deepening actually helps pruning
 				moves[i].value = negaMax(game, depth, depthCount, alpha, beta, color);
-				game.undoMove(moves[i].move);
+				game.takeMove(moves[i].move);
 
 			}
 			depthCount++;
 			//explore the best moves first
-			game.sortMoves(moves);
+			sortMoves(moves);
 		}
 
 		return moves[0].move;
 	}
 
 	//color keeps track of the player. It is 1 if we are the player, -1 if the opponent is the player
-	private int negaMax(Game game, int depth, int depthCount, int alpha, int beta, int color){
+	private static int negaMax(Board game, int depth, int depthCount, int alpha, int beta, int color){
 
 		//if we are at the end of the game or have reached the iterative deepening depth, evaluate the state of the board and return a reward
-		if(game.isOver() || depth == depthCount){
-			return color * game.evaluate(depth);
+		//Note: Maybe factor depth into the evaluation in the future?
+		if(game.state == BoardState.X_WON || game.state == BoardState.O_WON || depth == depthCount){
+			return color * Eval.evaluate(game);
 		}
 
 		//keeps track of the value of the best move we've found so far. Starts at -10000 so that the first value will override it
 		int max = -10000;
 
 		//explores each node in the tree
-		for(MoveAndValue possibleMove: game.getAvailableMoves()){
+		for(Move possibleMove: game.generateMoves()){
 
 			//these three lines make a move, recurivsely call the negamax on the new board state, and then un-make the move. This allows us to get the value of the move without
 			//"making" the move on our actual board
-			game.makeMove(possibleMove.move);
+			game.makeMove(possibleMove);
 			int negaMax_value = -negaMax(game, depth +1, depthCount, -beta, -alpha, -color);
```

Comment:
Looks good. I've got some questions about depth + 1 and depthCount - and how you implemented iterative deepening (I don't see you gradually incrementing depth?), but we can talk about those later. Typically, you pass into negamax a depth, then decrement it until that is zero. That's when you know you've gone as deep as the iterative deepening allows. Then, you increment that depth count, and start again.